### PR TITLE
refactor the getMaxResource method

### PR DESCRIPTION
**Pull request by nlu90**
As mentioned in #1161, put the getMaxResource refactor into a separate PR.

The new getMaxRequiredResource will find the max requirement for cpu, ram and disk from all the required container resources.

MesosScheduler and AuroraScheduler will use the new method to decide their resource allocation to each container.

### CODE REVIEW

**Comment by objmagic** [comment]
👍

**Comment by billonahill** [comment]
LGTM

@avflor and @ashvina you were also doing some similar refactors lately. Would you please review.

**Comment by avflor** [comment]
@nlu90 @billonahill The change looks good to me. The only difference is that in the original code Aurora takes only memory into account when computing the max resources whereas Mesos considers all the resources. I think it is more correct to use all resources as done in this PR.

**Comment by nlu90** [comment]
@avflor yeah, as we discussed in a previous refactor PR #1079 to take cpus and disks into consideration.

**Comment by kramasamy** [comment]
@nlu90 - if everything looks good, can you merge it?

File: heron/spi/src/java/com/twitter/heron/spi/utils/SchedulerUtils.java [hunk 1]
```diff
@@ -438,4 +438,20 @@ public static boolean curlAndExtractPackage(
...
+    PackingPlan.Resource maxResource = new PackingPlan.Resource(0, 0, 0);
+
+    for (PackingPlan.ContainerPlan entry : packingPlan.containers.values()) {
+      PackingPlan.Resource resource = entry.resource;
+      maxResource.ram = Math.max(maxResource.ram, resource.ram);
+      maxResource.cpu = Math.max(maxResource.cpu, resource.cpu);
+      maxResource.disk = Math.max(maxResource.disk, resource.disk);
+    }
+
+    return maxResource;
```
Comment (ashvina) [inline heron/spi/src/java/com/twitter/heron/spi/utils/SchedulerUtils.java]: Once the max resource is found, is there a need to update the `PackingPlan`? The resource requirement suggested by the`PackingPlan` could be different from actual allocation after this step. I am not sure if `PackingPlan` is used by any other component after the scheduler.

**Comment by ashvina** [comment]
The change looks good to me. It would be great if someone could provide some clarity on the question I posted above.

File: heron/spi/src/java/com/twitter/heron/spi/utils/SchedulerUtils.java (hunk 1 above)
Comment (nlu90) [inline heron/spi/src/java/com/twitter/heron/spi/utils/SchedulerUtils.java]: Scheduler can update the PackingPlan based on its need. But here, the util method only helps to find the maximum cpu/ram/disk resource requirement.

**Pull request merged**